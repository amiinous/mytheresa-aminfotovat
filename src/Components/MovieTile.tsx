import styled from 'styled-components/native'
import React, { useCallback } from 'react'
import MYImage from '@/Components/UIKit/Image/MYImage'
import { StyleProp, StyleSheet, ViewStyle } from 'react-native'
import Variables from '@/Theme/Variables'
import MYTouchableOpacity from '@/Components/UIKit/MYTouchableOpacity'
import { navigate } from '@/Navigator/utils'

const { dark } = Variables.Colors

interface Props {
  movie: Movie
  /**
   * The width style is the least required key.
   */
  style?: StyleProp<ViewStyle>
}

const MovieTile = ({ movie, style }: Props) => {
  const onPress = useCallback(() => navigate('MovieDetails', { movie }), [])
  return (
    <Container onPress={onPress} style={[styles.cardShadow, style]}>
      <PosterImage source={{ uri: movie?.poster_path }} />
    </Container>
  )
}

export default React.memo(
  MovieTile,
  (prevProps, nextProps) => prevProps.movie?.id === nextProps.movie?.id,
)

const Container = styled(MYTouchableOpacity)`
  border-radius: 8px;
  aspect-ratio: 0.5;
  elevation: 10;
  shadow-color: ${dark};
`
const PosterImage = styled(MYImage)`
  flex: 1;
  border-radius: 8px;
  overflow: hidden;
`

/**
 * I read the code of `css-to-react-native` library (https://github.com/styled-components/css-to-react-native/blob/master/src/transforms/boxShadow.js)
 * and as it always return the `shadowOpacity ` as 1 and I wanted it to be 0.5 (it was important to me!) I did not use styled-components to pass box shadow styles
 */
const styles = StyleSheet.create({
  cardShadow: {
    shadowOffset: { width: -2, height: 4 },
    shadowOpacity: 0.5,
    shadowRadius: 4,
  },
})
