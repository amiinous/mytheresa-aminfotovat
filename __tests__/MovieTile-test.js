import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import MovieTile from '@/Components/MovieTile'

describe('MovieTile Snapshot test ', () => {
  it('renders correctly', () => {
    expect(shallow(<MovieTile />)).toMatchSnapshot()
  })
})
