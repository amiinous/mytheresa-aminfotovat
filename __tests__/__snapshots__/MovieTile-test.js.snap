// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile Snapshot test  renders correctly 1`] = `
<Styled(_default)
  onPress={[Function]}
  style={
    Array [
      Object {
        "shadowOffset": Object {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      },
      undefined,
    ]
  }
>
  <Styled(MYImage)
    source={
      Object {
        "uri": undefined,
      }
    }
  />
</Styled(_default)>
`;
